packages/build-tools/src/utils/expoUpdates.ts

```diff
@@ -96,6 +100,38 @@ export async function configureExpoUpdatesIfInstalledAsync(
     ctx.logger.warn(
       "If you're using conditional app configs, e.g. depending on an environment variable, make sure to set the variable in eas.json or configure it with EAS Secret."
     );
+    if (ctx.metadata?.fingerprintSources && resolvedFingerprintSources && resolvedRuntimeVersion) {
+      const differences = diffFingerprints(
+        {
+          hash: ctx.metadata.runtimeVersion,
+          sources: ctx.metadata.fingerprintSources as FingerprintSource[],
+        },
+        {
+          hash: resolvedRuntimeVersion,
+          sources: resolvedFingerprintSources as FingerprintSource[],
+        }
+      );
+      ctx.logger.warn('Differences between local and resolved fingerprint:');
+      ctx.logger.warn(
+        JSON.stringify(
+          differences,
+          (key, value) => {
+            if (key === 'contents') {
+              try {
+                const item = JSON.parse(value);
+                return item;
+              } catch {
+                return value;
+              }
+            }
+            return value;
+          },
+          ' '
+        )
+      );
+    } else {
+      ctx.logger.warn(`Could not find fingerprint sources`);
```

Comment:
we should maybe consider if runtimeVersion.policy!=fingerprint case here. in this case i guess the `ctx.metadata.runtimeVersion` would be [`file:fingerprint`](https://github.com/expo/expo/blob/88b74f7ff00d37b2e44b1fa6e6dc764cc028b1d4/packages/%40expo/config-plugins/src/utils/Updates.ts#L16)

Reply:
Changed the logic here, we only compare fingerprint sources if both local and eas have them, and no warning is printed if they don't exist.